browser/ai_chat/page_content_fetcher_browsertest.cc

```diff
@@ -0,0 +1,109 @@
+
+namespace {
+
+constexpr char kEmbeddedTestServerDirectory[] = "text_recognition";
+
+}  // namespace
+
+class PageContentFetcherBrowserTest : public InProcessBrowserTest {
+ public:
+  PageContentFetcherBrowserTest()
+      : https_server_(net::EmbeddedTestServer::TYPE_HTTPS) {}
+
+  void SetUpOnMainThread() override {
+    mock_cert_verifier_.mock_cert_verifier()->set_default_result(net::OK);
+    host_resolver()->AddRule("*", "127.0.0.1");
+    content::SetupCrossSiteRedirector(&https_server_);
+
+    brave::RegisterPathProvider();
+    base::FilePath test_data_dir;
+    test_data_dir = base::PathService::CheckedGet(brave::DIR_TEST_DATA);
+    test_data_dir = test_data_dir.AppendASCII(kEmbeddedTestServerDirectory);
+    https_server_.ServeFilesFromDirectory(test_data_dir);
+
+    ASSERT_TRUE(https_server_.Start());
+  }
+
+  void SetUpCommandLine(base::CommandLine* command_line) override {
+    InProcessBrowserTest::SetUpCommandLine(command_line);
+#if BUILDFLAG(ENABLE_TEXT_RECOGNITION)
+    command_line->AppendSwitch(::switches::kEnablePixelOutputInTests);
+#endif
+    mock_cert_verifier_.SetUpCommandLine(command_line);
+  }
+
+  void SetUpInProcessBrowserTestFixture() override {
+    InProcessBrowserTest::SetUpInProcessBrowserTestFixture();
+    mock_cert_verifier_.SetUpInProcessBrowserTestFixture();
+  }
+
+  void TearDownInProcessBrowserTestFixture() override {
+    mock_cert_verifier_.TearDownInProcessBrowserTestFixture();
+    InProcessBrowserTest::TearDownInProcessBrowserTestFixture();
+  }
+
+  void NavigateURL(const GURL& url) {
+    content::WebContents* contents =
+        browser()->tab_strip_model()->GetActiveWebContents();
+    ASSERT_TRUE(ui_test_utils::NavigateToURL(browser(), url));
+    ASSERT_TRUE(WaitForLoadStop(contents));
+  }
+
+  void FetchPageContent(const std::string& expected_text,
+                        bool expected_is_video) {
+    base::RunLoop run_loop;
+    ai_chat::FetchPageContent(
+        browser()->tab_strip_model()->GetActiveWebContents(),
+        base::BindLambdaForTesting([&run_loop, &expected_text,
+                                    &expected_is_video](std::string text,
+                                                        bool is_video) {
+          ASSERT_EQ(expected_text, base::TrimWhitespaceASCII(
+                                       text, base::TrimPositions::TRIM_ALL));
+          ASSERT_EQ(expected_is_video, is_video);
+          run_loop.Quit();
+        }));
+    run_loop.Run();
+  }
+
+ protected:
+  content::ContentMockCertVerifier mock_cert_verifier_;
+  net::test_server::EmbeddedTestServer https_server_;
+};
+
+IN_PROC_BROWSER_TEST_F(PageContentFetcherBrowserTest, FetchPageContent) {
+  NavigateURL(https_server_.GetURL("a.com", "/image.html"));
+  FetchPageContent("this is the way", false);
+}
+
+#if BUILDFLAG(ENABLE_TEXT_RECOGNITION)
+IN_PROC_BROWSER_TEST_F(PageContentFetcherBrowserTest,
+                       FetchPageContentViaTextExtraction) {
```

Comment:
we also need a test case that a supported host fails `FetchPageContentViaTextExtraction` and fallback to normal `FetchPageContent`

Reply:
Discussed in Slack that this isn't needed because of no way to not get a view